### PR TITLE
language(typescript) actionTypes.js -> actionTypes.ts

### PR DESCRIPTION
**Pull request by Calinteodor**
1. Converted actionTypes to typescript
2. Updated eslintignore with actionTypes.ts
3. Updated contribution guidelines
4. Converted display name functions.js to functions.ts

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "target": "es5",
```
Comment (saghul) [inline tsconfig.json]: Make it es6 pl

File: react/features/base/lib-jitsi-meet/actions.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ import {
     LIB_INIT_ERROR,
     LIB_WILL_DISPOSE,
     LIB_WILL_INIT
-} from './actionTypes';
+} from './actionTypes.ts';
```
Comment (saghul) [inline react/features/base/lib-jitsi-meet/actions.js:15]: I thought we would not need this?

File: react/features/base/lib-jitsi-meet/index.js [hunk 3]
```diff
@@ -26,5 +26,5 @@ export const JitsiTrackErrors = JitsiMeetJS.errors.track;
 export const JitsiTrackEvents = JitsiMeetJS.events.track;
 
 export * from './actions';
-export * from './actionTypes';
+export * from './actionTypes.ts';
```
Comment (saghul) [inline react/features/base/lib-jitsi-meet/index.js:29]: Ditto.

File: react/features/base/media/middleware.js [hunk 4]
```diff
@@ -35,12 +35,7 @@ import {
     SET_VIDEO_MUTED,
     SET_VIDEO_UNMUTE_PERMISSIONS
 } from './actionTypes';
-import {
-    setAudioMuted,
-    setCameraFacingMode,
-    setScreenshareMuted,
-    setVideoMuted
-} from './actions';
+import { setAudioMuted, setCameraFacingMode, setScreenshareMuted, setVideoMuted } from './actions';
```
Comment (saghul) [inline react/features/base/media/middleware.js:38]: Pl undo this change.

File: react/features/display-name/components/native/DisplayNameLabel.tsx [hunk 5]
```diff
@@ -1,43 +1,39 @@
...
+} from '../../../base/participants/functions.js';
+import { connect } from '../../../base/redux/functions.js';
 
 import styles from './styles';
 
-type Props = {
+/**
+ * Renders a label with the display name of the on-stage participant.
+ */
+class DisplayNameLabel extends React.Component<{
```
Comment (saghul) [inline react/features/display-name/components/native/DisplayNameLabel.tsx:15]: Let's please create a Props type definition, similar to what was there before, but TS. It should make the code changes smaller and IMHO it's a simpler way forward.

File: react/features/display-name/components/native/DisplayNameLabel.tsx [hunk 6]
```diff
@@ -68,13 +64,12 @@ class DisplayNameLabel extends Component<Props> {
  * @returns {{
  * }}
  */
-function _mapStateToProps(state: Object, ownProps: Props) {
-    const { participantId, contained } = ownProps;
-    const participant = getParticipantById(state, participantId);
+function _mapStateToProps(state: Object, ownProps) {
```
Comment (saghul) [inline react/features/display-name/components/native/DisplayNameLabel.tsx:67]: IIRC in TS state here would be `object` or `any`

File: react/features/display-name/components/web/DisplayNameBadge.tsx [hunk 7]
```diff
@@ -1,22 +1,16 @@
-// @flow
-
 import { makeStyles } from '@material-ui/core/styles';
-import React from 'react';
-
-type Props = {
+import * as React from 'react';
```
Comment (saghul) [inline react/features/display-name/components/web/DisplayNameBadge.tsx:2]: why does the import need to be this way now?

File: react/features/display-name/components/web/DisplayNameBadge.tsx [hunk 8]
```diff
@@ -1,22 +1,16 @@
...
 
-    /**
-     * The name to be displayed within the badge.
-     */
-    name: string
-}
 
 const useStyles = makeStyles(theme => {
+    // @ts-ignore
```
[the remark this answers is not in the document]
Comment (saghul) [reply react/features/display-name/components/web/DisplayNameBadge.tsx:6]: what error do you get?

[the remark this answers is not in the document]
Comment (saghul) [reply react/features/display-name/components/web/DisplayNameBadge.tsx:6]: Maybe we need https://www.npmjs.com/package/@types/material-ui ?

File: react/features/lobby/actions.any.js [hunk 9]
```diff
@@ -219,7 +219,6 @@ export function startKnocking() {
         sendLocalParticipant(state, membersOnly);
 
         membersOnly.joinLobby(localParticipant.name, localParticipant.email);
-        dispatch(setLobbyMessageListener());
```
Comment (saghul) [inline react/features/lobby/actions.any.js:222]: Unrelated change.

File: react/features/recording/actions.any.js [hunk 10]
```diff
@@ -21,7 +21,12 @@ import {
     SET_SELECTED_RECORDING_SERVICE,
     SET_STREAM_KEY
 } from './actionTypes';
-import { getRecordingLink, getResourceId, isSavingRecordingOnDropbox, sendMeetingHighlight } from './functions';
+import {
```
Comment (saghul) [inline react/features/recording/actions.any.js:24]: Unrelated change.

File: tsconfig.json [hunk 11]
```diff
@@ -1,11 +1,14 @@
 {
+    "include": ["react/features/**/*.ts"],
```
[the remark this answers is not in the document]
Comment (saghul) [reply tsconfig.json:2]: I think we need to yeah. Does it not fail at all?

File: webpack.config.js [hunk 12]
```diff
@@ -220,7 +224,11 @@ function getConfig(options = {}) {
 
                 // Webpack defaults:
                 '.js',
-                '.json'
+                '.json',
+
+                // Typescript:
+                '.tsx',
```
Comment (saghul) [inline webpack.config.js:230]: Is this necessary?

**Review by saghul — CHANGES_REQUESTED** [review]
Some minor changes. Did you also test RN?

**Comment by saghul** [comment]
Also pl rebase, there seems to be a conflict now.

File: tsconfig.json (hunk 11 above)
[the remark this answers is not in the document]
Comment (Calinteodor) [reply tsconfig.json:2]: It doesn't.

File: react/features/base/lib-jitsi-meet/actions.js (hunk 2 above)
Comment (Calinteodor) [reply react/features/base/lib-jitsi-meet/actions.js:15]: If I remove this, the build fails.

File: react/features/display-name/components/web/DisplayNameBadge.tsx (hunk 7 above)
Comment (Calinteodor) [reply react/features/display-name/components/web/DisplayNameBadge.tsx:2]: Added in tsconfig "allowSyntheticDefaultImports": true, that I think is going to fix this.

File: react/features/display-name/components/web/DisplayNameBadge.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (Calinteodor) [reply react/features/display-name/components/web/DisplayNameBadge.tsx:6]: TS2339: Property 'text01' does not exist on type 'Palette

File: webpack.config.js (hunk 12 above)
Comment (Calinteodor) [reply webpack.config.js:230]: I thought it will be useful for components, regarding the fact that they will have an .tsx extension.

Comment (saghul) [reply webpack.config.js:230]: I don't think we need to add it because ts-loader will handle those files, this is for custom stuff like our `.web.js`

File: react/features/base/lib-jitsi-meet/actions.js (hunk 2 above)
Comment (saghul) [reply react/features/base/lib-jitsi-meet/actions.js:15]: Then we need to see why, the extension should not be needed at all.

Comment (Calinteodor) [reply react/features/base/lib-jitsi-meet/actions.js]: Updated. I think the problem was in an another place.

File: webpack.config.js (hunk 12 above)
Comment (saghul) [reply webpack.config.js:230]: 👍

File: react/features/display-name/components/native/DisplayNameLabel.tsx (hunk 5 above)
Comment (Calinteodor) [reply react/features/display-name/components/native/DisplayNameLabel.tsx]: Updated

**Review by saghul — APPROVED** [review]
